kyuubi-server/src/main/scala/org/apache/kyuubi/server/KyuubiTHttpFrontendService.scala

```diff
@@ -278,7 +278,11 @@ final class KyuubiTHttpFrontendService(
     val realUser = getShortName(Option(SessionManager.getUserName).getOrElse(req.getUsername))
     // using the remote ip address instead of that in proxy http header for authentication
     val ipAddress: String = SessionManager.getIpAddress
-    val sessionUser: String = getProxyUser(req.getConfiguration, ipAddress, realUser)
+    val sessionUser: String = if (req.getConfiguration == null) {
```

Comment:
Maybe move this check into `getProxyUser` method ?
`getProxyUser` checks the config value first and returns the `realUser` as default.  This config check could be added as a prior condition.

Reply:
My original intention here is to avoid `null` propagation as much as possible

Reply:
SGTM